scanners/ios/podfile.go

```diff
@@ -250,13 +255,61 @@ func GetWorkspaceProjectMap(podfilePth string, projects []string) (map[string]st
 	}, nil
 }
 
+func (podfileParser podfileParser) podfilelockPath(podfileDir string) (string, error) {
+	podfileLockPth := filepath.Join(podfileDir, "Podfile.lock")
+	if exist, err := pathutil.IsPathExists(podfileLockPth); err != nil {
+		return "", fmt.Errorf("failed to check if Podfile.lock exist, error: %s", err)
+	} else if !exist {
+		podfileLockPth = filepath.Join(podfileDir, "podfile.lock")
+		if exist, err := pathutil.IsPathExists(podfileLockPth); err != nil {
+			return "", fmt.Errorf("failed to check if podfile.lock exist, error: %s", err)
+		} else if !exist {
+			podfileLockPth = ""
+		}
+	}
+
+	return podfileLockPth, nil
+}
+
+func (podfileParser podfileParser) cocoapodsVersion(podfileLockPth string) (string, error) {
+	cocoapodsVersion := ""
+
+	if podfileLockPth != "" {
+		version, err := GemVersionFromGemfileLock("cocoapods", podfileLockPth)
+		if err != nil {
+			return "", fmt.Errorf("failed to read cocoapods version from %s, error: %s", podfileLockPth, err)
+		}
+		cocoapodsVersion = version
+	}
+
+	return cocoapodsVersion, nil
+}
+
+func (podfileParser podfileParser) fixPodfileQuotation(podfilePth string) error {
+	podfileContent, err := fileutil.ReadStringFromFile(podfilePth)
+	if err != nil {
+		return fmt.Errorf("failed to read podfile (%s), error: %s", podfilePth, err)
+	}
+
+	podfileContent = strings.Replace(podfileContent, `‘`, `'`, -1)
+	podfileContent = strings.Replace(podfileContent, `’`, `'`, -1)
+	podfileContent = strings.Replace(podfileContent, `“`, `"`, -1)
+	podfileContent = strings.Replace(podfileContent, `”`, `"`, -1)
+
+	if err := fileutil.WriteStringToFile(podfilePth, podfileContent); err != nil {
+		return fmt.Errorf("failed to apply Podfile quotation fix, error: %s", err)
+	}
+
+	return nil
+}
+
 // MergePodWorkspaceProjectMap ...
 // Previously we separated standalone projects and workspaces.
 // But pod workspace-project map may define workspace which is not in the repository, but will be created by `pod install`.
 // Related project should be found in the standalone projects list.
 // We will create this workspace model, join the related project and remove this project from standlone projects.
 // If workspace is in the repository, both workspace and project should be find in the input lists.
-func MergePodWorkspaceProjectMap(podWorkspaceProjectMap map[string]string, standaloneProjects []xcodeproj.ProjectModel, workspaces []xcodeproj.WorkspaceModel) ([]xcodeproj.ProjectModel, []xcodeproj.WorkspaceModel, error) {
+func (podfileParser podfileParser) MergePodWorkspaceProjectMap(podWorkspaceProjectMap map[string]string, standaloneProjects []xcodeproj.ProjectModel, workspaces []xcodeproj.WorkspaceModel) ([]xcodeproj.ProjectModel, []xcodeproj.WorkspaceModel, error) {
```

Comment:
Why does this belong to podfileParser? As far as I can see it deos not use any proprty from it.

Reply:
Nice catch 👍